libs/@hashintel/petrinaut/src/views/Editor/components/DiagnosticsPanel/diagnostics-panel.tsx

```diff
@@ -0,0 +1,404 @@
+import { css } from "@hashintel/ds-helpers/css";
+import { useCallback, useMemo, useRef, useState } from "react";
+import { FaChevronDown, FaChevronRight } from "react-icons/fa6";
+import ts from "typescript";
+
+import { useCheckerContext } from "../../../../state/checker-provider";
+import { useEditorStore } from "../../../../state/editor-provider";
+import { useSDCPNContext } from "../../../../state/sdcpn-provider";
+
+/**
+ * Formats a TypeScript diagnostic message to a readable string
+ */
+function formatDiagnosticMessage(
+  messageText: string | ts.DiagnosticMessageChain,
+): string {
+  if (typeof messageText === "string") {
+    return messageText;
+  }
+  return ts.flattenDiagnosticMessageText(messageText, "\n");
+}
+
+// Position offsets (accounting for sidebar padding/margins)
+const MIN_HEIGHT = 100;
+const MAX_HEIGHT = 600;
+const LEFT_SIDEBAR_WIDTH = 344; // 320px + 24px padding
```

Comment:
Can we pull the sidebar width from where it's defined? And the padding from wherever that is?

Reply:
I tried to centralize all panel-related stuff together, including some styles, but ended up with bugs on styles. 

If that is okay for you I'll add this on top of the PR fixing PandaCSS, or directly in the one that contains the BottomPanel (DiagnosticsPanel is already removed in my new PR)